Odata-docs/client/using-blazor-wasm-with-odata-client.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "Blazor Wasm with OData Client"
+description: "How to use blazor with odata client"
+
+author: ElizabethOkerio
+ms.author: elokerio
+ms.date: 1/18/2021
+ms.topic: article
+ 
+---
+# How To Use Blazor Web Assembly with OData Client 
+**Applies To**: [!INCLUDE[appliesto-odataclient](../includes/appliesto-odataclient-v7.md)]
+
+## Getting Started
+Start by creating a Blazor WebAssembly project.
+
+Open Visual Studio and click on "Create a new project" then select the "Blazor App template" and click "Next". Configure your project by providing the project name and the location path then click "Create". On the "Create Project" Page that appears, specify the target framework that you want to use and select "Blazor WebAssembly App" then click "Create". A blazor web assembly project will be created like below. 
+
+![SAMPLE BLAZOR WEB ASSEMBLY PROJECT](../assets/2021-02-02-blazor-wasm-app-sample.png)
+
+There are two ways in which you can use OData Client with blazor web assembly: 
+
+* You can use the Microsoft OData Connected Service
+* You can use plain-old C# objects (POCOs)
+
+## Using Microsoft OData Connected Service
+
+Add OData Connected service to the project you just created above. 
+
+To do so, ensure that you have the Microsoft OData Connected Service extension installed. Refer to this link [Microsoft OData Connected Service](/odata/connectedservice/getting-started/) on how to get started with Microsoft OData Connected Service.
+Right click on your project and add a connected service. Select the Microsoft OData Connected Service that you installed and use the Wizard to generate proxy classes for the OData endpoint that you want to use.
+
+Once all this is done, Open any of the Blazor razor pages that you want to display data from an OData endpoint on and include the code below on the `OnInitializedAsync` method.  
+
+```c#
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataserviceContect = new Default.Container(serviceRoot);
+dataserviceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataserviceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your razor pages.  
+
+## Using plain-old C# objects (POCOs)
+
+If you do not wish to use Microsoft Odata Connected Service or any code generation tools, then you create the `DataServiceContext` like below: 
+
+```c#
+public partial class ODataServiceContext : DataServiceContext
```

Comment:
I think it would be helpful to mention which namespace to include in order to get `DataServiceContext`.